packages/pnpm/src/main.ts

```diff
@@ -78,8 +78,12 @@ export default async function run (inputArgv: string[]) {
     argv: { remain: string[], cooked: string[], original: string[] },
   }
   try {
+    // When we just want to print the location of the global bin directory,
+    // we don't need the write permission to it. Related issue: #2700
+    const globalDirShouldAllowWrite = cmd === 'root'
```

Comment:
Shouldn’t this’ve been `cmd !== 'root'`?

Reply:
indeed. Seems like you are right

Reply:
fixed in master